Index: libgnomesu-1.0.0/tools/Makefile.am
===================================================================
--- libgnomesu-1.0.0.orig/tools/Makefile.am
+++ libgnomesu-1.0.0/tools/Makefile.am
@@ -1,7 +1,7 @@
 bin_PROGRAMS = gnomesu
 
 INCLUDES =\
-	-DGETTEXT_PACKAGE="\"gnomesu\""\
+	-DGETTEXT_PACKAGE="\"$(GETTEXT_PACKAGE)\""\
 	-DLOCALEDIR="\"$(datadir)/locale\""\
 	-DGNOMESU_VERSION="\"$(VERSION)\""\
 	-DGNOMESU_PREFIX="\"$(prefix)\""\
